### ISSUE TITLE
Error in training 

### ISSUE THREAD

**Issue by Erizu91** [Zhaoyi-Yan/Shift-Net_pytorch#132]
Hello, when I try to train I get this error ...

[Errno 99] Cannot assign requested address
on_close() takes 1 positional argument but 3 were given

How can a solve it ?
Thanks

**Comment by Zhaoyi-Yan** [comment]
try your `display_id=0` to disable the visualization during training. Maybe it is the incompatibility of visdom?

**Comment by Erizu91** [comment]
Thanks, now it works but when I finish the training I can't see the results by entering the URL.

**Comment by Zhaoyi-Yan** [comment]
try `display_port=9001`, if it still occurs an error, post the error message here.

**Comment by Erizu91** [comment]
Hi, I made the change you suggested in options/train_options.py but it keeps sending me to address 8097.

**Comment by Erizu91** [comment]
Hello, I'm still waiting for your help

**Comment by Zhaoyi-Yan** [comment]
Are you sure when you change https://github.com/Zhaoyi-Yan/Shift-Net_pytorch/blob/master/options/train_options.py#L12 to something like `9001`, it still sends you address 8097 even if your running script does not contains `display_port` item?

**Comment by Erizu91** [comment]
Yes, I changed the port number and retrained the code but it still appears that port number 8097 is not found.

**Comment by Erizu91** [comment]
Hi again, I retrained the code and I still can't see the results, I get the following message in the training ...

Could not connect to Visdom server. 
 Trying to start a server....
Command: /usr/bin/python3 -m visdom.server -p 8097 &>/dev/null &
create web directory ./log/exp/web...
[Errno 99] Cannot assign requested address

**Comment by Zhaoyi-Yan** [comment]
Maybe you are running the code on a server which contains no screens, you may try https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix , and ask for help there.
